pallets/circuit-clock/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+#![feature(associated_type_defaults)]
+//! <!-- markdown-link-check-disable -->
+//! # Account Manager pallet
+//! </pre></p></details>
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::type_complexity)]
+#![allow(clippy::too_many_arguments)]
+
+pub use crate::pallet::*;
+use frame_support::{
+    pallet_prelude::Weight,
+    traits::{Currency, Get},
+};
+
+pub use t3rn_primitives::{
+    abi::{GatewayABIConfig, Type},
+    common::RoundInfo,
+    protocol::SideEffectProtocol,
+    ChainId, GatewayGenesisConfig, GatewayType, GatewayVendor,
+};
+use t3rn_primitives::{
+    account_manager::AccountManager, executors::Executors, treasury::Treasury, EscrowTrait,
+};
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+mod weights;
+
+pub type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+// Definition of the pallet logic, to be aggregated at runtime definition through
+// `construct_runtime`.
+#[frame_support::pallet]
+pub mod pallet {
+    // Import various types used to declare pallet in scope.
+    use super::*;
+    use frame_support::{pallet_prelude::*, traits::ReservableCurrency};
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::traits::Zero;
+    use t3rn_primitives::{
+        circuit_clock::{CircuitClock, ClaimableArtifacts},
+        transfers::EscrowedBalanceOf,
+    };
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The overarching event type.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type Currency: ReservableCurrency<Self::AccountId>;
+
+        type WeightInfo: weights::WeightInfo;
+
+        /// Type providing some time handler
+        type Time: frame_support::traits::Time;
+
+        #[pallet::constant]
+        type EscrowAccount: Get<Self::AccountId>;
```

Comment:
hmm, should the clock have access to the escrow account? I don't think this is used here.

Reply:
yeah we'd need to decide what accounts distribute the rewards at claim. for now T::EscrowAccount keeps the rewards until the claim, not the biggest fan